api/v1alpha1/extension_types.go

```diff
@@ -74,6 +74,9 @@ type ExtensionSpec struct {
 	// paused controls the management state of the extension. If the extension is paused, it will be ignored by the extension controller.
 	Paused bool `json:"paused,omitempty"`
 
+	// NoopDelete controls what is done with the associated resources. If set to true they will not be deleted.
+	NoopDelete bool `json:"noopdelete,omitempty"`
```

Comment:
Nit: We could add validation to this with KB markers, setting it to optional, so we don't worry about the default value, then directly assign it to App's spec.

Reply:
done :)